UEBPrep/app/models/user.rb

```diff
@@ -34,4 +34,24 @@ def self.from_omniauth(auth)
       user.password = Devise.friendly_token[0,20]
     end
   end
+
+  def self.find_for_twitter_oauth(auth, signed_in_resource=nil)
+    user = User.where(:provider => auth.provider, :uid => auth.uid).first
+    if user
+      return user
+    else
+      registered_user = User.where(:email => auth.uid + "@twitter.com").first
```

Comment:
I thought of doing this work around FB too. I think we should just wait until we have the updated user model so we can create users via a username/password, the username which we can pull directly from twitter/facebook.